### PR TITLE
[PICS_V2_SUPPORT] Endpoints changes in order to support certification mode and mandatory Test and suites

### PR DESCRIPTION
**Pull request by rquidute**
### What changed
Updated the backend endpoints in order to support certification mode and mandatory Test and suites. 

### Related issue
https://github.com/project-chip/certification-tool/issues/290

### Testing
Unit test passing
Screenshot 2024-06-21 at 12 37 25

### CODE REVIEW

**Review by antonio-amjr — APPROVED** [review]

File: app/pics_applicable_test_cases.py [hunk 1]
```diff
@@ -40,15 +43,37 @@ def applicable_test_cases_list(pics: PICS) -> PICSApplicableTestCases:
...
+    # Add first the mandatories test cases
+    applicable_tests.extend(applicable_mandatories_tests)
+    # Add the remaining test cases
+    applicable_tests.extend(applicable_remaining_tests)
 
     logger.debug(f"Applicable test cases: {applicable_tests}")
     return PICSApplicableTestCases(test_cases=applicable_tests)
+
+
+def __append_test_cases(
```
Comment (ccruzagralopes) [inline app/pics_applicable_test_cases.py:62]: I think `__append_test_cases` suggests that the function will modify some list by appending the test cases, but it doesn't modify an existing object, it returns a new list.
```suggestion
def __applicable_test_cases(
```

File: test_collections/matter/sdk_tests/support/python_testing/models/python_test_models.py [hunk 2]
```diff
@@ -28,9 +28,11 @@ class PythonTestType(Enum):
     # - PythonTestType.NO_COMMISSIONING: test cases that follow the expected template
     #   but don't have a commissioning first step
     # - PythonTestType.LEGACY: test cases that don't follow the expected template
+    # - PythonTestType.MANDATORY: test cases that are mandatories
```
Comment (ccruzagralopes) [inline test_collections/matter/sdk_tests/support/python_testing/models/python_test_models.py:31]:
```suggestion
    # - PythonTestType.MANDATORY: test cases that are mandatory
```

File: test_collections/matter/sdk_tests/support/python_testing/models/python_test_parser.py [hunk 3]
```diff
@@ -32,6 +32,13 @@
 
 FunctionDefType = Union[ast.FunctionDef, ast.AsyncFunctionDef]
 
+mandatories_python_tcs_public_id = [
```
Comment (ccruzagralopes) [inline test_collections/matter/sdk_tests/support/python_testing/models/python_test_parser.py:35]:
```suggestion
mandatory_python_tcs_public_id = [
```

File: test_collections/matter/sdk_tests/support/python_testing/models/python_test_parser.py [hunk 4]
```diff
@@ -165,10 +172,14 @@ def __parse_test_case(
     # - PythonTestType.NO_COMMISSIONING: test cases that follow the expected template
     #   but don't have a commissioning first step
     # - PythonTestType.LEGACY: test cases that don't follow the expected template
+    # - PythonTestType.MANDATORY: Mandatories test cases
```
Comment (ccruzagralopes) [inline test_collections/matter/sdk_tests/support/python_testing/models/python_test_parser.py:175]:
```suggestion
    # - PythonTestType.MANDATORY: Mandatory test cases
```

**Review by gladystonfranca — APPROVED** [review]

**Review by ccruzagralopes — APPROVED** [review]

**Pull request merged**